### PR TITLE
chore: update the OpportunityCard component to show the icon for Foxy

### LINKED ISSUE #1540: FOXy token displayed for FOX staking opportunity
### Overview

Currently, when a user has staked their FOX for FOXy the opportunity in the DeFi section shows FOX token.  After a user has staked their FOX this should turn to the pink FOXy token (as the user now holds FOXy).

### References and additional details

This will be in the web code base.

Notes from engineering: the icon needs to be showing the asset for Fox Yieldy (FOXy) It can be found in state with the `assetId: eip155:1/erc20:0xdc49108ce5c57bc3408c3a5e95f3d864ec386ed3`

The icon should be found a the URL:
https://raw.githubusercontent.com/shapeshift/lib/main/packages/asset-service/src/generateAssetData/ethTokens/icons/foxy-icon.png

### Acceptance Criteria

A/C: FOXy is the token displayed for the staked FOX opportunity in the DeFi tab

### Need By Date

n/a

### Screenshots/Mockups

Screen Shot 2022-04-20 at 1 25 15 PM

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Estimated effort

2 hours

### Sponsor / Stakeholder

@FOX_McCloud

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### PR DESCRIPTION
**Pull request by nejos97**
## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #1540 
## Risk

You will now see the Foxy icon instead Fox in the defi page.

## Testing

Go to the defi page and see if the 'FOX token staking' icon is indeed Foxy's. 

## Screenshots (if applicable)
Screenshot 2022-06-02 at 01-16-57 ShapeShift DAO

### CODE REVIEW

File: src/pages/Defi/components/OpportunityCard.tsx [hunk 1]
```diff
@@ -13,7 +13,9 @@ import {
 import { cosmosChainId } from '@shapeshiftoss/caip'
 import { DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
 import { EarnOpportunityType } from 'features/defi/helpers/normalizeOpportunity'
+import { FoxAssetId, FoxyIconUrl } from 'plugins/foxPage/foxPage'
```
Comment (0xdef1cafe) [inline src/pages/Defi/components/OpportunityCard.tsx:16]: please don't import these from this plugin folder - plugins are supposed to be used in isolation, and data from them shouldn't be consumed outside of them

File: src/plugins/foxPage/foxPage.tsx [hunk 2]
```diff
@@ -42,6 +42,8 @@ export enum FoxPageRoutes {
 
 export const FoxAssetId = 'eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d'
 export const FoxyAssetId = 'eip155:1/erc20:0xdc49108ce5c57bc3408c3a5e95f3d864ec386ed3'
+export const FoxyIconUrl =
```
Comment (0xdef1cafe) [inline src/plugins/foxPage/foxPage.tsx:45]: please do not add this here

File: src/pages/Defi/components/OpportunityCard.tsx [hunk 3]
```diff
@@ -50,6 +52,9 @@ export const OpportunityCard = ({
   const asset = useAppSelector(state => selectAssetById(state, assetId))
   const { cosmosStaking } = useModal()
   const isCosmosStaking = chainId === cosmosChainId
+  const icon = useMemo(() => {
```
Comment (0xdef1cafe) [inline src/pages/Defi/components/OpportunityCard.tsx:55]: please do not put asset specific logic in this component, we need to see a generic solution that is extensible in future

the guidance i gave to a previous contributor was as follows

make a function that takes the assetId, and returns an override asset icon url

don't hardcode the URLs - get them from the asset service

use a object of the original asset id, to the override asset id, then get the icon from the override asset

if there isn't an override, just use the original icon

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

**Review by 0xApotheosis — APPROVED** [review]
🍭

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by 0xdef1cafe** [comment]
merging as the cypress failure is known and not critical

**Pull request merged**